purduesigbots/pros-cli
# ✨ Setup Autocomplete

#### Summary:

- Adds a command to setup autocomplete for bash, zsh, fish, pwsh, and powershell
- Adds a custom PowerShell script to register an argument completer


Example:
```bash
pros setup-autocomplete bash
```

#### Motivation:
Since we upgraded to click 8 in #307 it is possible to use click's shell completion feature. This will allow users to quickly select commands in the supported shells.

##### References (optional):
Closes #138 

#### Test Plan:

- [x] bash (Ubuntu WSL)
- [x] zsh (Ubuntu WSL)
- [x] fish (Ubuntu WSL)
- [x] pwsh (Ubuntu WSL)
- [x] bash (Windows)
- [ ] zsh (Windows)
   - Autocomplete doesn't work
- [x] pwsh (Windows)
- [x] powershell (Windows)
- [x] bash (macOS)
- [x] zsh (macOS)
- [x] fish (macOS)
- [ ] pwsh (macOS)

1. Run `pros setup-autocomplete {shell}`
2. Start typing a pros command, then press your autocomplete key (default should be double press TAB)
3. Ensure a completion is shown, example below for fish shell on linux
![image](https://github.com/purduesigbots/pros-cli/assets/34776435/029dadec-b34a-47cd-806a-5a4e8c98f888)
